### PR TITLE
Passphrase manager

### PR DESCRIPTION
**Pull request by lucksus**
Screenshot from 2019-03-15 18-05-08
- [ ] I have added a summary of my changes to the changelog

### CODE REVIEW

**Comment by maackle** [comment]
Will this eventually address the event of a power cycle, where the machine can't join the network again without human input?

**Comment by lucksus** [comment]
Hey @neonphog, I would love to have you have a quick look over this. Note that I'm currently using an insecure `SecBuf` because of the size constraint currently still present for secure ones. I think we need to change this before beta.

**Comment by thedavidmeister** [comment]
> Note that I'm currently using an insecure SecBuf

mm i wondered about that

File: conductor_api/src/conductor/passphrase_manager.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        let pm = PassphraseManager {
+            passphrase_cache: Arc::new(Mutex::new(None)),
+            passphrase_service,
+            last_read: Arc::new(Mutex::new(Instant::now())),
+            timeout_kill_switch: kill_switch_tx,
+        };
+
+        let pm_clone = pm.clone();
+
+        thread::spawn(move || loop {
```
Comment (neonphog) [inline conductor_api/src/conductor/passphrase_manager.rs]: I think we're going to need to do an audit of system threads at some point... they should probably mostly be moved into some kind of pool executor scheme.

**Comment by neonphog** [comment]
> Note that I'm currently using an insecure SecBuf

https://github.com/holochain/holochain-rust/issues/1017 strikes again!

File: conductor_api/src/conductor/passphrase_manager.rs (hunk 1 above)
Comment (lucksus) [reply conductor_api/src/conductor/passphrase_manager.rs]: :+1:

**Pull request merged**